spancontext.go

```diff
@@ -218,6 +218,10 @@ func (c *SpanContext) SetHTTPRequest(req *http.Request) {
 		Name:     destinationServiceURL.String(),
 		Resource: destinationServiceResource,
 	})
+	c.SetServiceTarget(ServiceTargetSpanContext{
+		Type: "http",
```

Comment:
According to https://github.com/elastic/apm/blob/main/specs/agents/tracing-spans-service-target.md#field-values:

> span.context.service.target.type should have the same value as span.subtype and fallback to span.type.

This would mean, for example, Elasticsearch spans would have `service.target.type: elasticsearch` (which is desirable). According to the pseudocode, we should only set Type if it is empty - and then, only to the span subtype or type.

What I'd suggest is we leave it empty here, and then set the default in `Span.End()`

Reply:
Note: I was planning to do the fallback mechanism in a separate PR: https://github.com/elastic/apm-agent-go/pull/1294

> This would mean, for example, Elasticsearch spans would have service.target.type: elasticsearch (which is desirable). According to the pseudocode, we should only set Type if it is empty - and then, only to the span subtype or type.

The current approach was mostly based on what `destination.resource` was doing. In particular, Elasticsearch would overwrite the service target fields (and the old destination resource) because it is calling `SetServiceTarget` after `SetHTTPRequest`.
Another option is to move the destination resource and service target fields outside of `SetHTTPRequest` and leave the responsibility of setting them to the caller.

> What I'd suggest is we leave it empty here, and then set the default in Span.End()

I think this approach won't work: `span.End` will only set the service target fields (or destination resource) if those fields weren't explicitly set (see the PR linked above for the code, it's based on what destination resource was doing). We could add another toggle/field to differentiate whether `span.End` should overwrite the fields but I feel like it would make the logic less robust. 

WDYT ?